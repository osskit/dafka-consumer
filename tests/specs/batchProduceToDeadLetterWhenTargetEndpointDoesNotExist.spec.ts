import type {Orchestrator} from '../testcontainers/orchestrator.js';
import {start} from '../testcontainers/orchestrator.js';
import {produce} from '../services/produce.js';
import {topicRoutes} from '../services/topicRoutes.js';
import delay from 'delay';
import {consume} from '../services/consume.js';
import {getOffset} from '../services/getOffset.js';

describe('tests', () => {
    let orchestrator: Orchestrator;

    beforeEach(async () => {
        orchestrator = await start(
            {
                KAFKA_BROKER: 'kafka:9092',
                MONITORING_SERVER_PORT: '3000',
                GROUP_ID: 'test',
                TARGET_BASE_URL: 'http://mocks:8080',
                TOPICS_ROUTES: topicRoutes([{topic: 'foo', targetPath: '/consume'}]),
                DEAD_LETTER_TOPIC: 'dead',
                TARGET_PROCESS_TYPE: 'batch',
            },
            ['foo', 'dead']
        );
    }, 5 * 60 * 1000);

    afterEach(async () => {
        if (!orchestrator) {
            return;
        }
        await orchestrator.stop();
    });

    it('batch produce to dead letter when target endpoint does not exist', async () => {
        //act
        await produce(orchestrator, {
            topic: 'foo',
            messages: [{value: JSON.stringify({data: 'foo'})}],
        });
        await delay(5000);

        //assert
        await expect(getOffset(orchestrator.kafkaClient, 'foo')).resolves.toBe(1);
        await expect(consume(orchestrator.kafkaClient, 'dead')).resolves.toMatchSnapshot();
    });
});
