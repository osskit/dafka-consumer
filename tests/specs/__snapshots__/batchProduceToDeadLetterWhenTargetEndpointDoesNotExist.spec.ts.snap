// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests batch produce to dead letter when target endpoint does not exist 1`] = `
{
  "headers": {
    "original-topic": "foo",
    "x-group-id": "test",
    "x-response-body": "No response could be served as there are no stub mappings in this WireMock instance.",
    "x-response-status-code": "404",
  },
  "value": {
    "data": "foo",
  },
}
`;
